lib/goldiloader/active_record_patches.rb

```diff
@@ -107,9 +107,19 @@ module AssociationPatch
     end
 
     def auto_include?
-      # We only auto include associations that don't have in-memory changes since the
-      # Rails association Preloader clobbers any in-memory changes
-      !loaded? && target.blank? && eager_loadable?
+      # No auto-include when the association is not eager loadable
```

Comment:
Do we need this logic for non-through associations? If not, should we move this fix to the special handling we have for middle association of a through associations [here](https://github.com/salsify/goldiloader/blob/master/lib/goldiloader/active_record_patches.rb#L180) since it's not clear why we'd want to overwrite an already loaded association?

Reply:
I think you are right, because in the non through cases the associations of the peers would already have been loaded before a modification could have been made.

I moved the code over there. I also think we don't need the additional `blank?` check and removed it.